### ISSUE TITLE
Multiple tasks of the same type generated for a single stack analyses request

### ISSUE THREAD

**Issue by samuzzal-choudhury** [openshiftio/openshift.io#1704]
During load testing with 50 concurrent stack-analyses requests, I have observed that a single request triggered multiple stackaggregator and recommender tasks with different task Ids. Because of this, the server could not return the response for that request and the load test had to be hard stopped.

Further, checking the database also revealed the behaviour-

```
coreapi=> select worker, worker_id from worker_results where external_request_id='0a7223a36c3e4782990beb1c3b406d5f';
       worker        |              worker_id               
---------------------+--------------------------------------
 GraphAggregatorTask | b361fc20-8fa6-4301-b062-7ca7adab9050
 recommendation_v2   | ab1fbcb9-08c2-43e1-b9ad-25cbcd0c711e
 stack_aggregator_v2 | 62ecdb5b-90bd-49b4-b270-5b3b3ba9b8db
 recommendation_v2   | 59c8b293-a4d5-4151-bea3-5b0614088294
 stack_aggregator_v2 | e2456759-6702-4066-8d93-fceec0ded250
(5 rows)
```
As it looks, GraphAggregator task managed to generate two different queue items for the same request with different task ids.

**Comment by samuzzal-choudhury** [comment]
Changed the stack analyses related tasks queue to FIFO from standard. Although this issue reduced to great extent, it does occur at times -

```
coreapi=> select worker, worker_id, external_request_id from worker_results where external_request_id='9aa5cbe22dbd43629b9859cab5344e12';
       worker        |              worker_id               |       external_request_id        
---------------------+--------------------------------------+----------------------------------
 stack_aggregator_v2 | b54cc93d-3576-4f77-aa66-d0355837793f | 9aa5cbe22dbd43629b9859cab5344e12
 GraphAggregatorTask | 435d02f8-cd52-4489-b2a6-e82bd3961192 | 9aa5cbe22dbd43629b9859cab5344e12
 GraphAggregatorTask | 378b4079-4fe8-427a-86e8-13c947097d11 | 9aa5cbe22dbd43629b9859cab5344e12
 stack_aggregator_v2 | acd40693-415f-41a8-a3ce-dd666f17d145 | 9aa5cbe22dbd43629b9859cab5344e12
 recommendation_v2   | fbe7754d-05c2-4e73-a488-cd889a8c2750 | 9aa5cbe22dbd43629b9859cab5344e12
 recommendation_v2   | a9ab2ab9-a369-44db-9e21-2b14989f322e | 9aa5cbe22dbd43629b9859cab5344e12
(6 rows)
```

**Comment by miteshvp** [comment]
@samuzzal-choudhury - as we observed this was a behavior around how SQS queue works
- FIFO - exactly once delivery
- Standard - atleast once delivery

WIth Standard queues under load tests, workers got the same tasks delivered on many occassions. It reduced for FIFO queues but we still observed some tasks were delivered to more than one workers because of the [visibility_timeout](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-lifecycle.html) settings. 
TL;DR - if a worker does not process a given message during the visibility timeout period, the same message is delivered back to the queue to be picked up by other worker processes. Unfortunately, under load, workers were overwhelmed and some of the workers could not finish the task in the said time window. Hence, the delivery of messages multiple times. 
Having said that, with the introduction of new service, we are trying to de-couple online load from the sqs queue completely. 
You can close this issue if you like.

**Comment by krishnapaparaju** [comment]
@samuzzal-choudhury now the SQS been removed from user facing traffic, am closing this issue..
